portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,39 +31,51 @@
 import java.util.List;
 import com.google.gson.Gson;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private static class Comment {
-    private String email, comment;
-    public Comment(String email, String comment){
+    private final String email, comment, time;
+    public Comment(String email, String comment, String time){
       this.email = email;
       this.comment = comment;
+      this.time = time;
     }
   }
-  private List<Comment> comments = new ArrayList<Comment>();
-
+  private Query comments_query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static final String COMMENT_DATE_FORMAT = "MMM dd,yyyy HH:mm";
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    PreparedQuery results = datastore.prepare(comments_query);
+    List<Comment> comments = new ArrayList<Comment>();
+    for(Entity comment : results.asIterable()){
+      comments.add(makeComment(comment));
+    }
     response.setContentType("application/json;");
     response.getWriter().println(convertToJsonUsingGson(comments));
   }
   
+  private static Comment makeComment(Entity comment){
+    SimpleDateFormat sdf = new SimpleDateFormat();    
+    Date resultDate = new Date((Long)comment.getProperty("time"));
+    return new Comment( (String)comment.getProperty("email"), 
+                        (String)comment.getProperty("comment"),
+                        sdf.format(resultDate));
+  }
+  
   private static String convertToJsonUsingGson(List<Comment> comments) {
     Gson gson = new Gson();
     return gson.toJson(comments);
   }
   
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    comments.add(new Comment(
-      request.getParameter("user_email"), 
-      request.getParameter("user_comment")));
-    
     Entity commentEntity = new Entity("Comment");
     commentEntity.setProperty("email", request.getParameter("user_email"));
     commentEntity.setProperty("comment", request.getParameter("user_comment"));
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    commentEntity.setProperty("time", System.currentTimeMillis());    
```

Comment:
With time, i tend to suggest people be more explicit about the exact units, unless it's captured in the type. My suggestion here is to use "timestamp_millis" or "time_millis".